### DISCUSSION TITLE
Is it possible to fix the height of the items but let the width be variable ?

### DISCUSSION THREAD

**Discussion by belfortf** [DerYeger/yeger#178] [General]
Hi, I would like to know if it is possible to fix the height of the items but let the width be variable ? Similar to what https://react-photo-album.com/ can do. Thank you.

**Comment by DerYeger** [comment]
There's no need to use this component for dynamic-width rows.
CSS has this layout built-in, in the form of `flex` rows with `flex-wrap` enabled.

**Reply by belfortf** [reply]
Yes, but if you need to constrain the width so that the card doesn't get too long, the masonry will eventually break on the right. The component I mentioned changes the order of the pics in order to find the best placement overall.

**Reply by DerYeger** [reply]
If the elements are not responsive, i.e., cannot scale their own width to fit, no algorithm is able to guarantee a perfect match.
CSS `flex-wrap` is good enough in my opinion.
The row-based layout differs too much from the column-based implementation of this single component library, I'm not planning on adding this.
